### ISSUE TITLE
Performance of a model loading the same weights changes depending on the version of tensorflow 

### ISSUE THREAD

**Issue by Jon573** [keras-team/keras#15513]
**System information**.
- Have I written custom code (as opposed to using a stock example script provided in Keras): yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 20.04 LTS
- TensorFlow installed from (source or binary): Binary
- TensorFlow version (use command below): 2.4.3 & 2.6.0
- Python version: 3.6.13
- GPU model and memory: RTX2080ti
- Exact command to reproduce: model.load_weights() & model.predict()

**Describe the problem**.
I received weights of 3 keras models (ResNet50V2, EfficientNetB0, and VGG16) that had been trained with a version of tensorflow 2.x (I don't know and can't find out). I do know the weights were created with the keras model checkpoint callback. I created a new venv and pip installed version 2.6.0 and loaded the weights and tested out the performance on the task. It was completely different to expected performance and terrible. It occurred for all three networks. I spent a while trying to trouble shoot and eventually thought I would try with an older version (2.4.3), and performance was as expected.

**Describe the expected behavior**.
Performance should not be different for models loading the same weights under 2.4.3 and 2.6.0.

**Comment by Jon573** [comment]
Here is some quick code I can reproduce the issue with. Set the CHECKPOINT_PATH to where you want, set EVALUATE to False, and run with TensorFlow 2.4.3. You should get an accuracy of ~0.65 . Change EVALUATE to True and run with with TensorFlow 2.4.3 and you should get the same accuracy (no surprises there). Then, with EVALUATE still True, run with TensorFlow 2.6.0 and I get an accuracy of ~0.1 . 

```
import tensorflow as tf

CHECKPOINT_PATH = ''
EVALUATE = True

def get_dataset():
    (train_images, train_labels), (test_images, test_labels) = tf.keras.datasets.cifar10.load_data()
    train_images, test_images = tf.image.per_image_standardization(train_images) , tf.image.per_image_standardization(test_images)
    return (train_images, train_labels, test_images, test_labels)

def get_model(load_weights):
    model = tf.keras.models.Sequential()
    base_model = tf.keras.applications.EfficientNetB0(
                                            input_shape=(32,32,3),
                                            weights='imagenet',
                                            include_top = False,
                                            ) 
    model.add(base_model)
    model.add(tf.keras.layers.GlobalAveragePooling2D())
    model.add(tf.keras.layers.Dropout(0.5))
    model.add(tf.keras.layers.Dense(10, activation = 'softmax', dtype='float32'))

    model.compile(optimizer='adam',
                loss=tf.keras.losses.SparseCategoricalCrossentropy(),
                metrics=['accuracy'])
    
    if load_weights:
        model.load_weights(CHECKPOINT_PATH)
    
    return(model)

def train_model(model, train_images, train_labels,test_images, test_labels):
    model_checkpoint_callback = tf.keras.callbacks.ModelCheckpoint(
        filepath=CHECKPOINT_PATH,
        save_weights_only=True,
        monitor='val_accuracy',
        mode='max',
        save_best_only=True)
    model.fit(train_images, 
                train_labels, 
                epochs=1, 
                validation_data=(test_images, test_labels),
                callbacks=[model_checkpoint_callback]
                )

def evaluate(model, train_images, train_labels,test_images, test_labels):
    _, test_acc = model.evaluate(test_images,  test_labels, verbose=2)
    print(test_acc)

def main():
    data = get_dataset()
    if EVALUATE:
        model = get_model(load_weights = True)
        evaluate(model, *data)
    else:
        model = get_model(load_weights = False)
        train_model(model, *data)
        evaluate(model, *data)

if __name__ == '__main__':
    main()
```

**Comment by chunduriv** [comment]
@Jon573, In general, the models improve with more epochs of training. Did you checked performance of model by increasing the number of epochs from `1` to `10`?

**Comment by tomerk** [comment]
That's a really good catch, thanks @Jon573 !
I'm guessing it was probably https://github.com/tensorflow/tensorflow/commit/b6be9714e878a7dd0d1405bd7a83e021ba4b561a

But it looks like the release notes release.md was not updated at the time even though it should have been.
You're right this issue would probably be better filed against the base tensorflow repo:
https://github.com/tensorflow/tensorflow

I'm not sure the behavior will be changed if it was originally a fix of an incorrect behavior change. But, the team can at least try updating release.md for past versions (not sure if it would get updated on already-published release notes though).

**Comment by tomerk** [comment]
@chunduriv is there a good process to move the issue over or cc teammates from the other tf repo here? (E.g. rohan100jain@ )

**Comment by SuryanarayanaY** [comment]
Hello, Thank you for reporting an issue.

We're currently in the process of migrating the new Keras 3 code base from keras-team/keras-core to keras-team/keras.
Consequently, This issue may not be relevant to the Keras 3 code base . After the migration is successfully completed, feel free to reopen this Issue at keras-team/keras if you believe it remains relevant to the Keras 3 code base. If instead this Issue is a bug or security issue in legacy tf.keras, you can instead report a new issue at [keras-team/tf-keras](https://github.com/keras-team/tf-keras/issues), which hosts the TensorFlow-only, legacy version of Keras.

To know more about Keras 3, please read https://keras.io/keras_core/announcement/
